code/lib/run_resample.py

```diff
@@ -0,0 +1,245 @@
+
+import os
+import subprocess
+import click
+from os.path import abspath
+from os.path import dirname
+from os.path import join
+from os.path import realpath
+
+
+from bidsNighres.bidsNighres.bidsutils import get_bids_filter_config
+from bidsNighres.bidsNighres.bidsutils import get_dataset_layout
+from bidsNighres.bidsNighres.bidsutils import check_layout
+from bidsNighres.bidsNighres.bidsutils import init_derivatives_layout
+from bidsNighres.bidsNighres.utils import print_to_screen
+from bidsNighres.bidsNighres.utils import return_path_rel_dataset
+
+from bids import BIDSLayout
+
+# __version__ = open(join(dirname(realpath(__file__)), "version")).read()
+
+
+def run_resample(command, env={}):
+    merged_env = os.environ
+    merged_env.update(env)
+    process = subprocess.Popen(
+        command,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        shell=True,
+        env=merged_env,
+    )
+    while True:
+        line = process.stdout.readline()
+        line = str(line, "utf-8")[:-1]
+        print(line)
+        if line == "" and process.poll() != None:
+            break
+    if process.returncode != 0:
+        raise Exception("Non zero return code: %d" % process.returncode)
+
+
+@click.command()
+@click.option(
+    "--input-datasets",
+    help="""
+            The directory with the input dataset formatted according to the BIDS standard.
+            """,
+    type=click.Path(exists=True, dir_okay=True),
+    required=True,
+)
+@click.option(
+    "--output-location",
+    help="""
+            The directory where the output files should be stored.
+            If you are running group level analysis this folder should be prepopulated
+            with the results of the participant level analysis.
+            """,
+    type=click.Path(exists=False, dir_okay=True),
+    required=True,
+)
+# @click.option(
+#     "--resample-factor",
+#     help="""
+#             Factor by which the image will be resampled, for example  "2" means double the resoulution.
+#             """,
+#     type=click.IntRange(min=0.1, max=3, min_open=False, max_open=False, clamp=True),
+#     default=2,
+#     show_default=True,
+# )
+@click.option(
+    "--participant-label",
+    help="""
+            The label(s) of the participant(s) that should be analyzed. The label
+            corresponds to sub-<participant_label> from the BIDS spec
+            (so it does not include "sub-"). If this parameter is not
+            provided all subjects should be analyzed. Multiple
+            participants can be specified with a space separated list.
+            """,  # nargs ?
+    required=True,
+)
+@click.option(
+    "--bids-filter-file",
+    help="""
+            Path to a JSON file to filter input file
+            """,
+    default="",
+    show_default=True,
+)
+def main(
+    input_datasets,
+    output_location,
+    participant_label,
+    bids_filter_file,
+):
+
+    resampling_factor = 2
```

Comment:
this probably will have to turned into a parameter of the "run_resample" function